### PR TITLE
[SYCL][FPGA] Make pipe feature to be Intel extension

### PR DESCRIPTION
**Pull request by MrSidims**
pipe class was moved under intel namespace and it's definition
is now placed in fpga_extensions.hpp.

Old definition was temporary left in where it was before.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by sergey-semenov** [comment]
I'm not a fan of keeping duplicate code in sycl/pipes.hpp and sycl/intel/pipes.hpp, even if it's temporary. How about putting something like
`namespace cl {
namespace sycl {
using cl::sycl::intel::pipe;
}
}`
in sycl/pipes.hpp instead?

**Comment by sergey-semenov** [comment]
I would also argue that this commit does introduce some functional changes, I think the NFC tag should be reserved strictly for code refactoring/cleanup.

**Comment by MrSidims** [comment]
> I'm not a fan of keeping duplicate code in sycl/pipes.hpp and sycl/intel/pipes.hpp, even if it's temporary. How about putting something like
> `namespace cl { namespace sycl { using cl::sycl::intel::pipe; } }`
> in sycl/pipes.hpp instead?

Thanks for the idea.

File: sycl/include/CL/sycl/pipes.hpp [hunk 1]
```diff
@@ -10,78 +10,12 @@
 
 #include <CL/__spirv/spirv_ops.hpp>
 #include <CL/__spirv/spirv_types.hpp>
+#include <CL/sycl/intel/pipes.hpp>
```
Comment (sergey-semenov) [inline sycl/include/CL/sycl/pipes.hpp]: No need for the other includes with the latest change

Comment (MrSidims) [reply sycl/include/CL/sycl/pipes.hpp]: Actually, these is a need of these include. Otherwise cl::sycl::intel::pipe isn't known at this point.

Comment (sergey-semenov) [reply sycl/include/CL/sycl/pipes.hpp]: I meant that all includes BUT this one can go

Comment (MrSidims) [reply sycl/include/CL/sycl/pipes.hpp]: Oh, yes, indeed.

File: sycl/include/CL/sycl/pipes.hpp [hunk 2]
```diff
@@ -8,80 +8,11 @@
 
 #pragma once
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/pipes.hpp]: Why not remove this file completely? AFAIK users are not supposed to include this file directly anyway.

**Pull request merged**